app/models/event.rb

```diff
@@ -76,4 +83,41 @@ def coach_emails
   def permitted_audience_values
     %w[Students Coaches]
   end
+
+  def sponsors?(level = nil)
+    case level
+    when :gold then gold_sponsors.any?
+    when :silver then silver_sponsors.any?
+    when :bronze then bronze_sponsors.any?
+    when :standard then sponsors.any?
+    else sponsorships.any?
+    end
+  end
+
+  private
+
+  def duplicated_sponsors
+    @duplicated_sponsors ||= fetch_duplicated_sponsors
```

Comment:
I think changing the validations to no longer be per-level would mean this no longer needs to be memoized